lib/HTFeed/DataDenZipAudit.pm

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/perl
+package HTFeed::DataDenZipAudit;
+
+use strict;
+use warnings;
+use HTFeed::DBTools qw(get_dbh);
+use HTFeed::Volume;
+use Carp;
+use Log::Log4perl;
+
+# Procedural function for randomly choosing a volume to check.
+#
+# my $vol = HTFeed::DataDenZipAudit->choose();
+# my $audit = HTFeed::DataDenZipAudit->new(namespace => $vol->{namespace},
+#                                          objid => $vol->{objid},
+#                                          path => $vol->{path},
+#                                          version => $vol->{version});
+# my $result = $audit->run();
+sub choose {
+  my $path_prefix = shift || '/htdataden';
+
+  my $sql = 'SELECT COUNT(*) FROM feed_backups' .
+            " WHERE path LIKE '$path_prefix%'" .
+            ' AND deleted IS NULL AND saved_md5sum IS NOT NULL';
+  my $dbh = HTFeed::DBTools->get_dbh();
+  my ($rows) = $dbh->selectrow_array($sql);
+  my $random_row = int(rand($rows));
+  my ($namespace, $objid, $path, $version);
+  $sql = 'SELECT namespace,id,path,version FROM feed_backups' .
+         " WHERE path LIKE '$path_prefix%'" .
+         ' AND deleted IS NULL' .
+         ' AND saved_md5sum IS NOT NULL' .
+         ' LIMIT ?,1';
+  eval {
+    ($namespace, $objid, $path, $version) = $dbh->selectrow_array($sql, undef, $random_row);
+  };
+  if ($@) {
+    die "Database query failed: $@";
+  }
+  return { namespace => $namespace, objid => $objid,
+           path => $path, version => $version };
+}
+
+sub new {
+  my $class = shift;
+
+  my $object = {
```

Comment:
Generally in this code base I've used `$self` for the object we build in the constructor.